src/components/Carousel.tsx

```
@@ -1,26 +1,173 @@
-      </li>
-    </ul>
-
-    <button type="button">Prev</button>
-    <button type="button">Next</button>
-  </div>
-);
-
-export default Carousel;
+interface Props {
+  images: string[];
+  imageWidth: number;
+  newImageWidth: (width: number) => void;
+  frameSize: number;
+  newFrameSize: (size: number) => void;
+  currentStep: number;
+  newStep: (step: number) => void;
+  animationSpeed: number;
+  newSpeed: (speed: number) => void;
+  currentPosition: number;
+  newPosition: (position: number) => void;
+}
+
+export const Carousel: React.FC<Props> = ({
+  imageWidth,
+  newImageWidth,
+  frameSize,
+  newFrameSize,
+  images,
+  currentStep,
+  newStep,
+  animationSpeed,
+  newSpeed,
+  currentPosition,
+  newPosition,
+}) => {
+  return (
+    <div className="main">
+      <div className="Carousel">
+        <ul className="Carousel__list">
+          {images.map((src, index) => (
+            <li key={index}>
+              <img src={src} alt={`${index + 1}`} />
+            </li>
+          ))}
+        </ul>
+      </div>
+
+      <div className="Carousel__buttons">
+        <button
+          type="button"
+          onClick={() => {
+            const carouselList = document.querySelector(
+              '.Carousel__list',
+            ) as HTMLUListElement;
+            const maxNegative = 0;
+            const stepPx = currentStep * 130;
+
+            let nextPosition = currentPosition + stepPx;
+
+            if (nextPosition > maxNegative) {
+              nextPosition = maxNegative;
+            }
+
+            newPosition(nextPosition);
+            carouselList.style.transform = `translateX(${nextPosition}px)`;
+          }}
+        >
+          Prev
+        </button>
+
+        <button
+          type="button"
+          onClick={() => {
+            const carouselList = document.querySelector(
+              '.Carousel__list',
+            ) as HTMLUListElement;
+            const maxNegative = -((10 - frameSize) * imageWidth);
```

Issue: The calculation for `maxNegative` uses a hardcoded value of `10` for the number of images. This should use `images.length` to support a dynamic number of images and avoid errors if the images array changes.